code/modules/projectiles/guns/ballistic/launchers.dm

```diff
@@ -148,3 +148,51 @@
 	. = ..()
 	playsound(loc, "sparks", 75, 1, -1)
 	do_sparks(8, 3, usr)
+
+/obj/item/gun/ballistic/maintMusket
+	name = "maintenance musket"
+	desc = "A weapon with notoriously poor accuracy; it makes up for this by being quick and easy to smash together"
+	icon_state = "maintMusket"
+	item_state = "maintMusket"
+	mag_type = /obj/item/ammo_box/magazine/internal/cartridge
+	w_class = WEIGHT_CLASS_HUGE
+	weapon_weight = WEAPON_HEAVY
+	slot_flags = ITEM_SLOT_BACK
+	casing_ejector = FALSE
+	bolt_type = BOLT_TYPE_NO_BOLT
+	internal_magazine = TRUE
+	empty_indicator = FALSE
+	can_bayonet = TRUE
+	knife_x_offset = 27
+	knife_y_offset = 13
+	can_suppress = FALSE
+	force = 5
+	cartridge_wording = "cartridge"
+	fire_sound = 'sound/weapons/musketShot.ogg'
+	load_sound = 'sound/weapons/musketCock.ogg'
+	pin = /obj/item/firing_pin
+	var/reloading_active = FALSE
+
+/obj/item/gun/ballistic/maintMusket/afterattack()
+	. = ..()
+	magazine.get_round(FALSE)
+
+/obj/item/gun/ballistic/maintMusket/attackby(obj/item/A, mob/user, params)
+	
+	if(istype(A, /obj/item/ammo_casing/caseless/cartridge) || istype(A, /obj/item/kitchen/knife))
+		if(istype(A, /obj/item/kitchen/knife)) //Attach bayonet
+			..()
+			return
+		else if(reloading_active == TRUE)
+			to_chat(user, span_warning("You're already reloading it!"))
+			return
+		to_chat(user, span_notice("You start reloading the [src]."))
+		user.visible_message(span_warning("[user] starts reloading the [src]!"))
+		reloading_active = TRUE
+		if(!do_after(user, 5 SECONDS, user))
+			reloading_active = FALSE
+			user.balloon_alert(user, "You were interupted!")
+			return
+		..()
+		chamber_round()
+		reloading_active = FALSE
```

Comment:
there are so many things wrong with this part that it's hard for me to explain

Reply:
I made some changes. Did I miss anything?